client/app/dedicatedCloud/datacenter/drp/steps/common/confirmation/dedicatedCloud-datacenter-drp-confirmationStep.controller.js

```diff
@@ -0,0 +1,148 @@
+
+        if (state === this.DEDICATEDCLOUD_DATACENTER_DRP_STATUS.delivered) {
+          this.Alerter.success(`
+            ${this.$translate.instant('dedicatedCloud_datacenter_drp_confirm_create_success_part_one')} ${this.$translate.instant('dedicatedCloud_datacenter_drp_confirm_create_success_part_two', { email: this.email })}
+          `, 'dedicatedCloudDatacenterDrpAlert');
+        }
+      })
+      .catch((error) => {
+        this.Alerter.error(`${this.$translate.instant('dedicatedCloud_datacenter_drp_confirm_create_error')} ${_.get(error, 'data.message', error.message)}`, 'dedicatedCloudDatacenterDrpAlert');
+      })
+      .finally(() => {
+        this.isLoading = false;
+      });
+  }
+
+  storeZertoOptionOrderInUserPref(drpInformations, enableDrp) {
+    const drpInformationsToStore = {
+      drpInformations,
+      zertoOptionOrderId: enableDrp.orderId,
+      zertoOption: drpInformations.drpType === this.DEDICATEDCLOUD_DATACENTER_DRP_OPTIONS.ovh
+        ? this.DEDICATEDCLOUD_DATACENTER_DRP_ORDER_OPTIONS.zertoOption.ovh
+        : this.DEDICATEDCLOUD_DATACENTER_DRP_ORDER_OPTIONS.zertoOption.onPremise,
+    };
+
+    const { splitter } = this.DEDICATEDCLOUD_DATACENTER_ZERTO;
+    const [, ...[formattedServiceName]] = drpInformations.primaryPcc.serviceName.split(splitter);
+    const preferenceKey = `${this.DEDICATEDCLOUD_DATACENTER_ZERTO.title}_${formattedServiceName.replace(/-/g, '')}`;
+
+    return this.ovhUserPref.create(preferenceKey, drpInformationsToStore);
+  }
+
+  getOtherPccInformations() {
+    const primaryOptions = {
+      serviceName: _.get(this.drpInformations, 'primaryPcc.serviceName'),
+      datacenterId: _.get(this.drpInformations, 'primaryDatacenter.id'),
+    };
+
+    const secondaryOptions = {
+      serviceName: _.get(this.drpInformations, 'secondaryPcc.serviceName'),
+      datacenterId: _.get(this.drpInformations, 'secondaryDatacenter.id'),
+    };
+
+    return [primaryOptions, secondaryOptions]
+      .find(({ serviceName }) => serviceName !== this.$stateParams.productId);
+  }
+
+  regenerateZsspPassword() {
+    this.generatingPassword = true;
+    return this.DedicatedCloudDrp.regenerateZsspPassword(this.drpInformations)
+      .then(() => {
+        this.Alerter.success(`
+          ${this.$translate.instant('dedicatedCloud_datacenter_drp_confirm_create_success_part_two', { email: this.email })}
+        `, 'dedicatedCloudDatacenterDrpAlert');
+      })
+      .catch((error) => {
+        this.Alerter.error(`${this.$translate.instant('dedicatedCloud_datacenter_drp_confirm_zssp_password_regenerate_error')} ${_.get(error, 'message', '')}`, 'dedicatedCloudDatacenterDrpAlert');
+      })
+      .finally(() => {
+        this.generatingPassword = false;
+      });
+  }
+
+  deleteDrpModal() {
+    this.$uibModal.open({
+      template,
+      controller: 'DedicatedCloudDatacenterDrpConfirmationStepDeleteCtrl',
+      controllerAs: '$ctrl',
+    }).result
```

Comment:
You could use a modal with a `state` by applying `layout: 'modal'` on it

Reply:
Hmm, yes but I need the callback of `$uibModalInstance` when modal is closed